cmd/livepeer/livepeer.go

```diff
@@ -77,7 +77,7 @@ func main() {
 	ipfsPath := flag.String("ipfsPath", fmt.Sprintf("%v/.ipfs", usr.HomeDir), "IPFS path")
 	noIPFSLogFiles := flag.Bool("noIPFSLogFiles", false, "Set to true if log files should not be generated")
 	offchain := flag.Bool("offchain", false, "Set to true to start the node in offchain mode")
-	publicAddr := flag.String("publicAddr", "", "Public address that broadcasters can use to contact this node; may be an IP or hostname. If used, should match the on-chain ServiceURI set via livepeer_cli")
+	serviceAddr := flag.String("serviceAddr", "", "Transcoder only. Public address:port that broadcasters can use to contact this node; may be an IP or hostname. If used, should match the on-chain ServiceURI set via livepeer_cli")
```

Comment:
Yeah, there's a few somewhat conflicting goals here. 

* We already force HTTPS in the Service URI; it's added in implicitly by the CLI. The user doesn't have to specify a the `https` scheme when setting the ServiceURI, just ip:port . See https://github.com/livepeer/go-livepeer/blob/df3618534c37a4ef03ea1608e97ba7a9b0b468fc/cmd/livepeer_cli/wizard_transcoder.go#L47-L62
* `-serviceAddr` was used to try and maintain consistency with the `xxxAddr` scheme of the other options. We have rtmpAddr, httpAddr and cliAddr . Those only require an ip:addr , so we only require an ip:addr when setting serviceAddr. (This also matches how the CLI asks for the service URI before setting it on-chain.) Internally, we construct that into a full URI and parse that during the check at startup.
* The default ServiceURI port is already 4433. (At the time, we couldn't reuse 8935.) We can change this to 8935. Note that the 'https alternative' port, 8443, may be more friendly to some firewalls, if we're going to make a change. https://github.com/livepeer/go-livepeer/blob/df3618534c37a4ef03ea1608e97ba7a9b0b468fc/cmd/livepeer_cli/wizard_transcoder.go#L14
* Users will notice the CLI port changes if they are invoking the CLI endpoints (`/debug`, `/status` , `/manifestID`).

Given those choices, I think I'd rather keep the CLI port as-is, and maybe change the default RPC/HTTP port to 8443. That should buy us a little more compatibility with broadcasters.

Whatever we do, we just have to stress this difference in the documentation. Mostly that means communicating to our existing users. I can post a thorough explanation on the forums and the docs. Hopefully newcomers won't have trouble picking this up, since they don't have prior conditioning to break.

Reply:
Sure, set the default CLI to 7935 and RPC/HTTP to 8935 . https://github.com/livepeer/go-livepeer/pull/512/commits/d8223b265031242b1cb43f3260a1cc83d23769a8